components/TripForm/index.js

```diff
@@ -12,29 +12,56 @@ import {
 import ConfirmationMessage from "../ConfirmationMessage";
 import { useRouter } from "next/router";
 import { useState } from "react";
+import { mutate } from "swr";
 
+const CLOUD_NAME = process.env.NEXT_PUBLIC_CLOUDINARY_CLOUD_NAME;
+const UPLOAD_PRESET = "ml_default";
+
+async function uploadImage(file) {
+  const formData = new FormData();
+  formData.append("file", file);
+  formData.append("upload_preset", UPLOAD_PRESET);
+  const response = await fetch(
+    `https://api.cloudinary.com/v1_1/${CLOUD_NAME}/image/upload`,
+    {
+      method: "POST",
+      body: formData,
+    }
+  );
+  const { url, width, height } = await response.json();
+  return { url, width, height };
+}
+
+async function createPost(data) {
+  return await fetch("/api/trips", {
+    method: "POST",
+    body: JSON.stringify(data),
+    headers: {
+      "Content-Type": "application/json",
+    },
+  });
+}
 export default function TripForm() {
   const [letters, setLetters] = useState(150);
   const router = useRouter();
 
   async function handleSubmit(event) {
     event.preventDefault();
+    try {
+      const image = await uploadImage(event.target.image.files[0]);
 
-    const formData = new FormData(event.target);
-    const tripData = Object.fromEntries(formData);
-    const response = await fetch("/api/trips", {
-      method: "POST",
-      headers: {
-        "Content-Type": "application/json",
-      },
-      body: JSON.stringify(tripData),
-    });
-    if (!response.ok) {
-      console.error(response.status);
-      return;
-    }
+      const formData = new FormData(event.target);
+      const tripData = Object.fromEntries(formData);
+      await createPost({
+        ...tripData,
+        image,
+      });
 
-    router.push("/confirmation");
+      mutate("/api/trips");
```

Comment:
I believe you do not need to use `mutate()` here. You are redirecting the user back to the homepage via the confirmation page. On your homepage, you have your `TripList`, which is fetching your trips with `useSWR`, so the data should get automatically revalidated.